indico/modules/events/controllers/creation.py

```diff
@@ -57,16 +59,25 @@ def _get_form_defaults(self):
         if category is not None:
             tzinfo = timezone(category.timezone)
 
-        # try to find good dates/times
-        now = now_utc(exact=False)
-        start_dt = now + relativedelta(hours=1, minute=0)
-        if start_dt.astimezone(tzinfo).time() > time(18):
-            start_dt = tzinfo.localize(datetime.combine(now.date() + relativedelta(days=1), time(9)))
-        end_dt = start_dt + relativedelta(hours=2)
+        title = None
+        try:
+            cached_data = self._cache.get(request.args['event_uuid'])
+            title = cached_data['title']
+            start_dt = datetime.strptime(cached_data['start_dt'], "%Y-%m-%d %H:%M:%S")
+            start_dt = tzinfo.localize(start_dt)
+            end_dt = start_dt + relativedelta(minutes=int(cached_data['duration']))
+        except Exception:
```

Comment:
Please don't ever do that except in VERY exceptional cases where you really want to ignore all errors (or when you re-`raise` at the end)

Reply:
```python
prepared_data = self._get_prepared_data()
if prepared_data:
    tz, start_dt, end_dt, title = prepared_data
else:
    # try to find good dates/times
    ...
```

and a method like this:

```python
def _get_prepared_data(self):
    try:
        data = self._cache.get(request.args['event_uuid'])
    except KeyError:
        return None
    if not data:
        return None
    tz = data['timezone']
    start_dt = datetime.strptime(data['start_dt'], '%Y-%m-%d %H:%M')
    start_dt = timezone(tz).localize(start_dt)
    end_dt = start_dt + relativedelta(minutes=cached_data['duration'])
    return tz, start_dt, end_dt, title
```

Note: the timezone needs to be passed from newdle since we know the timezone there, but all we have here would be the default timezone which doesn't necessarily match the one from the newdle!

You could also move the parsing of the time string into the preparation endpoint and already store a datetime object in the cache entry.

Reply:
Thanks for explaining ;)

I've added a `_get_prepared_data` method and am now passing timezone info from Newdle and storing the parsed datetime in cache.